### ISSUE TITLE
FSDP: Mixed Precision follow up work

### ISSUE THREAD

**Issue by rohan-varma** [pytorch/pytorch#74515]
### 🚀 The feature, motivation and pitch

Tracking issue for follow up work for mixed precision (added in https://github.com/pytorch/pytorch/pull/74452):

- [x] Test local_state_dict checkpoint works with mixed precision. In particular we have to be careful about buffer casting.
- [x] For BatchNorm, initial solution will be to skip mixed precision training using auto_wrap API, user can also specify to disable mixed precision for batch norm via `wrap` API. https://github.com/pytorch/pytorch/issues/75478
- [x] In future, we would want to have some sort of flag to force inputs to fp32 for the batchnorm layer
- [x] Implement sharded grad scaler
- [x] Test that mixed precision works with amp
- [ ] FairScale comparison tests
- [x] Investigate whether separate stream is truly needed for CPU offload + MP: https://github.com/pytorch/pytorch/pull/74452#discussion_r833765789
- [x] Add additional tests for ShardedGradScaler and working with torch.cuda.amp.

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

**Comment by zrbcool** [comment]
any update about this?
